### PR TITLE
Migrate randomgen architecture disabling from OpenJ9 to OMR

### PR DESCRIPTION
**Pull request by fjeremic**
Issue: eclipse/openj9#3576

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -583,6 +583,48 @@ OMR::Z::CodeGenerator::CodeGenerator()
...
+            }
+
+         case 3:
+            {
+            _processorInfo.disableArch(TR_S390ProcessorInfo::TR_z13);
+            traceMsg(comp, "RandomGen: Disabling z13 processor architecture.");
+            break;
+            }
+
+         case 3:
```
Comment (Leonardo2718) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: I think this is supposed to be:
```suggestion
         case 4:
```

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -583,6 +583,48 @@ OMR::Z::CodeGenerator::CodeGenerator()
...
+            {
+            _processorInfo.disableArch(TR_S390ProcessorInfo::TR_z14);
+            traceMsg(comp, "RandomGen: Disabling z14 processor architecture.");
+            break;
+            }
+
+         case 4:
```
Comment (Leonardo2718) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: And here
```suggestion
         case 5:
```

File: compiler/z/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (Leonardo2718) [reply compiler/z/codegen/OMRCodeGenerator.cpp]: Unless the cases were supposed to start at 0...

**Comment by fjeremic** [comment]
Thanks for catching that @Leonardo2718. I might add at his point that it is unfortunate that most of the processor detection currently sits in OpenJ9, unrightfully so. I'm slowly working on pushing this code down into OMR.

[1] https://github.com/eclipse/openj9/blob/35c44ead49a53f8c0ad1ca3e63c18711d3a7b26f/runtime/compiler/env/ProcessorDetection.cpp#L854-L857

**Comment by Leonardo2718** [comment]
@genie-omr build zlinux,zos

**Comment by Leonardo2718** [comment]
> I might add at his point that it is unfortunate that most of the processor detection currently sits in OpenJ9, unrightfully so. I'm slowly working on pushing this code down into OMR.

Glad to hear your working on moving that code to OMR :smile: . Not having access to it is a major drawback right now.

**Pull request merged**